Training/python/Phase2/Training_WPloss.py

```diff
@@ -36,11 +36,16 @@
 
 class DeepTauModel(keras.Model):
```

Comment:
would it be possible to have a single `Training_CNN.py` shared between phase 2 and run 3 and set parameters (including the choice of the loss function) from the config?

Reply:
Ok, I removed the second file, and now the regular `Training.py` can be used for both the tau_crossentropy and the new loss.
One note however: I was never able to make the new loss function work without using eagerly evaluation, so now it's enabled only when using the new loss function:
https://github.com/cms-tau-pog/TauMLTools/blob/fb754c3f85a087563ac7c351080291bcd5474be9/Training/python/Phase2/Training.py#L619

Reply:
Can we merge code even further? What are the differences between the python code for Phase 2 and Run 3? As far as I see, there is no reason to diverge at the code level - I would keep the same code for Run 3, boosted and Phase 2, with the different configuration files for each case.

Reply:
You mean to merge `2018v1/Training_v0p1.py` and `Phase2/Training.py`? There are a few differences, which I can try to merge. What I notice at first glance is that the metrics for tensorboard are different (the list is hardcoded), because when using the new loss function with eagerly evaluation, reducing the number of metric helps to somewhat compensate for the longer runtime.

Reply:
yes, we can rename the merged file to `Training_CNN.py`. A list of active metrics can be moved to the config - it shouldn't be a problem.